carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/DriverPool.java

```diff
@@ -68,15 +68,16 @@ public static WebDriver getDriver() {
     }
 
     /**
-     * Get first registered driver from Pool.
+     * @deprecated (Obsolete method which will be removed in 6.0 core generation. Standard getDriver(..) should be reused!)
      * 
+     * Get first registered driver from Pool.
      * @return default WebDriver
      */
     @Deprecated
     public static WebDriver getExistingDriver() {
         ConcurrentHashMap<String, WebDriver> currentDrivers = getDrivers();
         if (currentDrivers.size() == 0) {
-            throw new RuntimeException("Unable to find exiting river in DriverPool!");
+            throw new RuntimeException("Unable to find exiting driver in DriverPool!");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://demo.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS00112)